stdlib/src/builtin/string.mojo

```diff
@@ -1971,6 +1972,115 @@ struct String(
             )
         return String(buf^)
 
+    @always_inline
+    fn _format[
+        *Ts: Stringable
+    ](
+        inout self,
+        args: VariadicPack[_, _, Stringable, Ts],
```

Comment:
I'm not sure variadic packs work in parameter contexts, this might be the issue.

Reply:
:+1: , this review can be implemented,
it would require to refine a previous review, which asked to do `String(self).format(args)`:
- `*args` cannot be passed to another function that takes `*args`, 
only to one that takes `VariadicPack`or `VariadicListMem`.

*(`str` with `VariadicPack` don't work in the parameter domain.)*

If we design it to work in the parameter domain,
both `StringLiteral` and `String` need an duplicate implementation that takes `*args`.
*(`str` works with "simple" `*args` in the parameter domain)*

Need an answer for this review, 
can both `.format()` be a duplicate of exactly the same logic?

Reply:
I think if you change the implementation in `StringLiteral` to just use `String.format`, you would not have to forward `*args`, and you could just get rid of `_format` completely.

Reply:
Yes, lets not forward `*args` but how because `String.format` need it?

Reply:
You can just implement `String.format`. Then `StringLiteral` can call it directly. If you're unsure, just address the other reviews, and we'll get back to this later.

Reply:
@laszlokindrat Sure, here is a commit !